app/lib/app_secrets.rb

```diff
@@ -0,0 +1,19 @@
+class AppSecrets
+  def self.[](key)
+    result = Vault.kv(namespace).read(key)&.data&.fetch(:value) if ApplicationConfig["VAULT_TOKEN"].present?
+    result ||= ApplicationConfig[key]
+
+    result
+  rescue Vault::VaultError
+    ApplicationConfig[key]
+  end
+
+  def self.[]=(key, value)
+    Vault.kv(namespace).write(key, value: value)
```

Comment:
should this bail as well if VAULT_TOKEN is not set?

Reply:
I wasn't sure how I wanted to handle this honestly bc I am not sure how/if I will use this in production. I thought about raising an error or returning nil without vault. I think this is going to evolve as our Vault usage evolves.